packages/grafana-ui/src/components/Icon/Icon.tsx

```diff
@@ -35,13 +35,17 @@ const getIconStyles = stylesFactory((theme: GrafanaTheme) => {
 });
 
 function getIconSubDir(name: IconName, type: string): string {
-  return name?.startsWith('gf-')
-    ? 'custom'
-    : alwaysMonoIcons.includes(name)
-    ? 'mono'
-    : type === 'default'
-    ? 'unicons'
-    : 'mono';
+  if (name?.startsWith('gf-')) {
+    return 'custom';
+  } else if (alwaysMonoIcons.includes(name)) {
+    return 'mono';
+  } else if (type === 'default') {
+    return 'unicons';
+  } else if (type === 'solid') {
+    return 'solid';
+  } else {
+    return 'mono';
+  }
```

Comment:
might make sense to add a few tests for this, just because it's in the public grafana-ui and susceptible to accidental breaking changes?